src/Interface/Decidable/Instance.agda

```diff
@@ -34,7 +35,16 @@ infix 0 ifᵈ_then_else_
 ifᵈ_then_else_ : (P : Set a) → ⦃ Dec P ⦄ → A → A → A
 ifᵈ P then t else f = if ⌊ ¿ P ¿ ⌋ then t else f
 
-Decidable²⇒Dec : {_~_ : A → A → Set a} → Decidable² _~_ → {x y : A} → Dec (x ~ y)
+Dec¹ : (A → Set a) → Set _
```

Comment:
These and many other improvements below were contributed by @omelkonian. See the branch [34-type-classes-orders-preferred-orestis-review](https://github.com/input-output-hk/formal-ledger-specifications/tree/34-type-classes-orders-preferred-orestis-review)